iceoryx_posh/doc/chunk_header.md

```diff
@@ -0,0 +1,218 @@
+# Summary
+
+Chunks are the transport capsules in iceoryx. They store data from a publisher as payload and are sent to one or more subscriber. Furthermore, there is some meta information which is stored alongside the payload, e.g. the size and the origin of the chunk. This data is composed in the `ChunkHeader` and located at the front of the chunk. Custom meta information can be added to extend the data from the `ChunkHeader` and tailor the chunk to specific use cases. While this makes the chunk layout more complex, this complexity would otherwise be distributed at different locations all over the code base, e.g. in the request/response feature. Additionally, the adjustments for the custom header makes arbitrary alignments of the payload trivial to implement.
+
+# Terminology
+
+| Name              | Description                                              |
+| :---------------- | :------------------------------------------------------- |
+| Chunk             | a piece of memory                                        |
+| Chunk Header      | contains meta information related to the chunk           |
+| Custom Header     | contains custom meta information, e.g. timestamps        |
+| Payload           | the user data                                            |
+| Back-Offset       | offset stored in front of the payload to calculate back to the chunk header |
+
+# Design
+
+## Considerations
+
+- it's not uncommon to record chunks for a later replay -> detect incompatibilities on replay
+- iceoryx runs on multiple platforms -> endianness of recorded chunks might differ
+- for tracing, a chunk should be uniquely identifiable -> store origin and sequence number
+- the chunk is located in the shared memory, which will be mapped to arbitrary positions in the address space of various processes -> no absolute pointer are allowed
+- aligning the `ChunkHeader` to 32 bytes will ensure that all member are on the same cache line and will improve performance
```

Comment:
Why 32 byte?

Reply:
historical reasons. Up until now the chunks are aligned to 32 bytes to ensure that we also support an alignment of up to 32 bytes for the payload. With this change, the alignment of the  payload can be specified, so it's not necessary anymore. It might help with performance if the `ChunkHeader` is not split at two cache lines and 32 is currently the size of the `ChunkHeader`. We might want to trade this for more memory efficiency in case we need to increase the `ChunkHeader` to e.g. 40 bytes, then the alignment would enforce a size of 64 bytes with 24 padding bytes. In this case it might be better to have an alignment of `8` bytes with a potential cost of distributing the `ChunkHeader` on multiple cache lines. This could also be a compile time option to shift the decision to the user.